pkg/controller/certificates/trigger/trigger_controller.go

```diff
@@ -160,6 +160,19 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 		return nil
 	}
 
+	// If we detect that the Certificate has a duplicate Secret name set, we don't trigger issuance.
```

Comment:
I don't think this comment adequately explains the strategy. I suggest something like this:

> It is possible for multiple Certificates to reference the same Secret and in that case,
> the duplicate Certificates will each be issued and store their version of the X.509 certificate in the target Secret,
> triggering the re-issuance of the other Certificate resources who's spec no longer matches what is in the Secret.
> This causes a flood of re-issuance attempts and overloads the Kubernetes API 
> and the API server of the issuing CA.  
> The problem is solved as follows:
> 1. Checking whether the target Secret exists and it is annotated with the name of this Certificate.
> 2. If yes, then this Certificate is re-issued.
> 3. If no, then this Certificate will be requeued to be re-reconciled every 3 minutes, so that it can be re-issued its secretRef field gets updated to a a unique Secret.
> 4. If the Secret does not exist, then the oldest Certificate which references it will be assumed to be the future owner.
> 5. If this Certificate is that future owner, it will be re-issued.
> 6. If not, then this Certificate will be requeued to be re-reconciled every 3 minutes, so that it can be re-issued its secretRef field gets updated to a a unique Secret.

Reply:
Please also  explain in a comment why `ownerRef` wasn't used for this purpose.

Reply:
done